src/components/SimulatorArea.tsx

```diff
@@ -33,18 +39,42 @@ const Canvas = styled('canvas', {
   touchAction: 'none',
 });
 
-export class SimulatorArea extends React.Component<SimulatorAreaProps> {
+export class SimulatorArea extends React.Component<SimulatorAreaProps, SimulatorAreaState> {
   private containerRef_: HTMLDivElement;
   private canvasRef_: HTMLCanvasElement;
 
   constructor(props: SimulatorAreaProps) {
     super(props);
-    this.state = {};
+    this.state = { 
+      loading: true,
+      loadingMessage: '',
+    };
   }
 
   componentDidMount() {
+    // after a few seconds of loading, show some messages to the user
+
+    // 10 second message: still going, might have fail
+    setTimeout(() => {
+      if (this.state.loading) {
+        this.setState({ loadingMessage: 'This process is taking longer than expected...' });
+      }
+    }, 10000);
```

Comment:
I think these timeouts should be even longer. The robot model itself is ~6 MB, so on slower connections, just downloading the robot model on first use will take a while

Reply:
At 20 Mbps, 6 MB is 2.4 seconds, and is around 48 seconds at 1 Mbps. I think US average for internet is >20 Mbps but it was hard to find exact numbers. How about 20 seconds for the first message, and 2 minutes for the second?

Reply:
Sounds good! Mostly thinking about schools with slow speeds, or large Botball events where wifi sometimes struggles